### PR TITLE
HAAR-1221: delete email domain test and other update

### PR DESCRIPTION
**Pull request by olatomoloju**
Good morning All, please could somebody help out with this delete email domain tests PR.

### CODE REVIEW

File: backend/controllers/deleteEmailDomain.js [hunk 1]
```diff
@@ -24,13 +23,32 @@ const deleteEmailDomainFactory = (deleteEmailDomainApi, listEmailDomainsUrl) =>
...
+            href: '#domainName',
+            text: 'Unauthorized to use the delete email domain functionality',
+          },
+        ]
+        stashStateAndRedirectToIndex(req, res, domainUnauthorizedError)
+      } else if (error.status === 403 && error.response && error.response.body) {
+        const domainRoleForbiddenError = [
+          {
+            href: '#domainName',
+            text: 'Forbidden to use the delete email domain functionality without the relevant role',
```
Comment (karenmillermoj) [inline backend/controllers/deleteEmailDomain.js:38]: Are these error conditions covered by authorisationMiddleware

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Review by karenmillermoj — APPROVED** [review]
lgtm

File: backend/controllers/deleteEmailDomain.js (hunk 1 above)
Comment (olatomoloju) [reply backend/controllers/deleteEmailDomain.js]: As discussed 403 is covered by the auth middleware but the other error scenarios will remain. Thanks Karen.

**Pull request merged**